src/test/pmem2_config/pmem2_config.c

```diff
@@ -346,6 +346,122 @@ test_set_directory_fd(const char *file)
 
+/*
+ * test_set_offset_too_large - test tries to set the offset which
+ * is too large
+ */
+static void
+test_set_offset_too_large(const char *unused)
+{
+	struct pmem2_config cfg;
+
+	/* let's try to set the offset which is too large */
+	size_t offset = (size_t)INT64_MAX + 1;
+	int ret = pmem2_config_set_offset(&cfg, offset);
+	UT_PMEM2_EXPECT_RETURN(ret, PMEM2_E_RANGE);
+}
+
+/*
+ * test_set_offset_no_multiple - test tries to set the offset which is not
+ * a multiple of Pagesize
+ */
+static void
+test_set_offset_no_multiple(const char *unused)
+{
+	struct pmem2_config cfg;
+
+	/* let's try to set the offset which is not a multiple of Pagesize */
+	size_t offset = Pagesize + 1;
+	int ret = pmem2_config_set_offset(&cfg, offset);
+	UT_PMEM2_EXPECT_RETURN(ret, PMEM2_E_UNALIGNED);
+}
+
+/*
+ * test_set_offset_success - test tries to successfully set the offset
+ */
+static void
+test_set_offset_success(const char *unused)
+{
+	struct pmem2_config cfg;
+
+	/* let's try to successfully set the offset */
+	size_t offset = Pagesize;
+	int ret = pmem2_config_set_offset(&cfg, offset);
+	UT_ASSERTeq(ret, 0);
+}
+
+/*
+ * test_set_length_too_large - test tries to set the length which
+ * is too large
+ */
+static void
+test_set_length_too_large(const char *unused)
+{
+	struct pmem2_config cfg;
+
+	/* let's try to set the length which is too large */
+	size_t length = (size_t)INT64_MAX + 1;
+	int ret = pmem2_config_set_offset(&cfg, length);
+	UT_PMEM2_EXPECT_RETURN(ret, PMEM2_E_RANGE);
+}
+
+/*
+ * test_set_length_no_multiple - test tries to set the length which is not
+ * a multiple of Pagesize
+ */
+static void
+test_set_length_no_multiple(const char *unused)
+{
+	struct pmem2_config cfg;
+
+	/* let's try to set the length which is not a multiple of Pagesize */
+	size_t length = Pagesize + 1;
+	int ret = pmem2_config_set_offset(&cfg, length);
+	UT_PMEM2_EXPECT_RETURN(ret, PMEM2_E_UNALIGNED);
+}
+
+/*
+ * test_set_length_success - test tries to successfully set the length
+ */
+static void
+test_set_length_success(const char *unused)
+{
+	struct pmem2_config cfg;
+
+	/* let's try to successfully set the offset */
+	size_t length = Pagesize;
+	int ret = pmem2_config_set_offset(&cfg, length);
+	UT_ASSERTeq(ret, 0);
+}
+
+/*
+ * test_set_length_max - test tries to successfully set maximum possible length
+ */
+static void
+test_set_length_max(const char *unused)
+{
+	struct pmem2_config cfg;
+
+	/* let's try to successfully set maximum possible length */
+	size_t length = (INT64_MAX / Pagesize) * Pagesize;
+	int ret = pmem2_config_set_offset(&cfg, length);
+	UT_ASSERTeq(ret, 0);
+}
+
+/*
+ * test_set_offset_max - test tries to successfully set maximum possible offset
+ */
+static void
+test_set_offset_max(const char *unused)
+{
+	struct pmem2_config cfg;
+
+	/* let's try to successfully set maximum possible offset */
+	size_t offset = (INT64_MAX / Pagesize) * Pagesize;
+	int ret = pmem2_config_set_offset(&cfg, offset);
+	UT_ASSERTeq(ret, 0);
+}
+
```

Comment:
I think you should at least add a test that maps something and verifies if it's of required length and offset in the file.